From cb6a3f62155468084a247dde5e3c563a2f7bcd3d Mon Sep 17 00:00:00 2001
From: Mike Tipton <mdtipton@codeaurora.org>
Date: Mon, 2 Nov 2020 15:27:58 -0800
Subject: [PATCH] NOUPSTREAM: ANDROID: clk: Enable writable debugfs files

Enable writable clock debugfs files which are required for debugging and
testing. This doesn't introduce security or safety issues for Android
since debugfs itself is already considered unsafe and mounting it isn't
allowed on user devices.

[CPNOTE: 06/07/21] Lee: Requested an upstreamable solution
[CPNOTE: 15/07/21] Lee: Upstream have repeatedly rejected making this a Kconfig option

Bug: 172264964
Change-Id: I68bd54cfe3765e2b3cbb208383c6117e662054b7
Signed-off-by: Mike Tipton <mdtipton@codeaurora.org>
Signed-off-by: Lee Jones <joneslee@google.com>
---
 drivers/clk/clk.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/clk/clk.c b/drivers/clk/clk.c
index 8a24567098fc..dce31ca5b160 100644
--- a/drivers/clk/clk.c
+++ b/drivers/clk/clk.c
@@ -3353,7 +3353,7 @@ static int clk_dump_show(struct seq_file *s, void *data)
 }
 DEFINE_SHOW_ATTRIBUTE(clk_dump);
 
-#undef CLOCK_ALLOW_WRITE_DEBUGFS
+#define CLOCK_ALLOW_WRITE_DEBUGFS
 #ifdef CLOCK_ALLOW_WRITE_DEBUGFS
 /*
  * This can be dangerous, therefore don't provide any real compile time
-- 
2.34.1

